streaming/stream_session.cc

```diff
@@ -45,10 +45,41 @@ static sstables::offstrategy is_offstrategy_supported(streaming::stream_reason r
     static const std::unordered_set<streaming::stream_reason> operations_supported = {
         streaming::stream_reason::bootstrap,
         streaming::stream_reason::replace,
+        streaming::stream_reason::removenode,
+        streaming::stream_reason::decommission,
+        streaming::stream_reason::repair,
+        streaming::stream_reason::rebuild,
     };
     return sstables::offstrategy(operations_supported.contains(reason));
 }
 
+class offstrategy_trigger {
```

Comment:
This class has nothing specific to streaming, has it?
It can be made avilable to all in a public header file..

Also, we should think how repair and streaming interact with compaction_group and later on with tablets.
This should be the context of triggering off-strategy compaction, not the whole table.

Reply:
It is specific to how streaming trigger off strategy compaction. It is not intended to be a public helper. It is better to stay inside .cc file.